starport/pkg/availableport/availableport.go

```diff
@@ -1,22 +1,34 @@
 package availableport
 
 import (
+	"fmt"
+	"math/rand"
 	"net"
+	"time"
 )
 
 // Find finds n number of unused ports.
 // it is not guaranteed that these ports will not be allocated to
 // another program in the time of calling Find().
 func Find(n int) (ports []int, err error) {
+	min := 44000
+	max := 55000
+
 	for i := 0; i < n; i++ {
-		ln, err := net.Listen("tcp", ":0")
-		if err != nil {
-			return nil, err
-		}
-		if err := ln.Close(); err != nil {
-			return nil, err
+		for {
+			rand.Seed(time.Now().UnixNano())
+			port := rand.Intn(max-min+1) + min
+
+			conn, err := net.Dial("tcp", fmt.Sprintf(":%d", port))
+			// if there is an error, this might mean that no one is listening from this port
+			// which is what we need.
+			if err == nil {
+				conn.Close()
+				continue
+			}
+			ports = append(ports, port)
```

Comment:
Maybe we could add a additional check to see if the error match a specific pattern. There might be other errors for `Dial` failing like the port open but not able to Dial with tcp